indi-maxdomeii/maxdomeii.cpp

```diff
@@ -643,54 +647,59 @@ bool MaxDomeII::ISNewNumber(const char *dev, const char *name, double values[],
             sprintf(cLog, "New home azimuth set: %lf", nVal);
             nHomeAzimuth              = nVal;
             nHomeTicks                = floor(0.5 + nHomeAzimuth * nTicksPerTurn / 360.0);
-            HomeAzimuthNP.s           = IPS_OK;
-            HomeAzimuthNP.np[0].value = nVal;
-            IDSetNumber(&HomeAzimuthNP, "%s", cLog);
+            HomeAzimuthNP.setState(IPS_OK);
+            // HomeAzimuthNP.[0].value = nVal;
+            LOGF_INFO("%s", cLog);
+            HomeAzimuthNP.apply();
             return true;
         }
         // Incorrect value.
-        HomeAzimuthNP.s = IPS_ALERT;
-        IDSetNumber(&HomeAzimuthNP, "Invalid home azimuth");
+        HomeAzimuthNP.setState(IPS_ALERT);
+        LOG_INFO("Invalid home azimuth");
+        HomeAzimuthNP.apply();
 
         return false;
     }
 
     // ===================================
     // Watch dog
     // ===================================
-    if (!strcmp(name, WatchDogNP.name))
+    if (WatchDogNP.isNameMatch(name))
     {
         double nVal;
         char cLog[255];
 
-        if (IUUpdateNumber(&WatchDogNP, values, names, n) < 0)
+        if (WatchDogNP.update(values, names, n) < 0)
             return false;
 
         nVal = values[0];
         if (nVal >= 0 && nVal <= 3600)
         {
             sprintf(cLog, "New watch dog set: %lf", nVal);
-            WatchDogNP.s           = IPS_OK;
-            WatchDogNP.np[0].value = nVal;
-            IDSetNumber(&WatchDogNP, "%s", cLog);
+            WatchDogNP.setState(IPS_OK);
+            // FixMe
```

Comment:
Also, pending.

Reply:
Same, this is an outdated file. It is already updated. i.e using WatchDogNP[0].setValue(nVal);